src/EditorFeatures/CSharpTest/CodeRefactorings/UseExplicitOrImplicitType/UseExplicitTypeRefactoringTests.cs

```diff
@@ -423,6 +423,462 @@ static void Main()
+        {
+            var code = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    static void Main()
+    {
+        [||]var v = """";
+        Action a = () => {
+            v = GetString();
+        };
+    }
+
+    static string GetString() => """";
+}";
+
+            var expected = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    static void Main()
+    {
+        string v = """";
+        Action a = () => {
+            v = GetString();
+        };
+    }
+
+    static string GetString() => """";
+}";
+
+            await TestInRegularAndScriptWhenDiagnosticNotAppliedAsync(code, expected);
+        }
+
+        [Fact]
+        public async Task TestNullabilityAssignment_Lambda4()
+        {
+            var code = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    static void Main()
+    {
+        var[||] v = """";
+        Action a = () => {
+            v = GetString();
+        };
+    }
+
+    static string? GetString() => null;
+}";
+
+            var expected = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    static void Main()
+    {
+        string? v = """";
+        Action a = () => {
+            v = GetString();
+        };
+    }
+
+    static string? GetString() => null;
+}";
+
+            await TestInRegularAndScriptWhenDiagnosticNotAppliedAsync(code, expected);
+        }
+
+        [Fact]
+        public async Task TestNullabilityAssignment_Property1()
+        {
+            var code = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    string S
+    {
+        get 
+        {
+            var[||] v = """";
+            v = GetString();
+            return v;
+        }
+    }
+
+    static void Main()
+    {
+    }
+
+    static string GetString() => """";
+}";
+
+            var expected = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    string S
+    {
+        get 
+        {
+            string v = """";
+            v = GetString();
+            return v;
+        }
+    }
+
+    static void Main()
+    {
+    }
+
+    static string GetString() => """";
+}";
+
+            await TestInRegularAndScriptWhenDiagnosticNotAppliedAsync(code, expected);
+        }
+
+        [Fact]
+        public async Task TestNullabilityAssignment_Property2()
+        {
+            var code = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    string? S
+    {
+        get 
+        {
+            var[||] v = """";
+            v = GetString();
+            return v;
+        }
+    }
+
+    static void Main()
+    {
+    }
+
+    static string? GetString() => null;
+}";
+
+            var expected = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    string? S
+    {
+        get 
+        {
+            string? v = """";
+            v = GetString();
+            return v;
+        }
+    }
+
+    static void Main()
+    {
+    }
+
+    static string? GetString() => null;
+}";
+
+            await TestInRegularAndScriptWhenDiagnosticNotAppliedAsync(code, expected);
+        }
```

Comment:
would like some tests in some some 'interesting' locations.  for example, a lambda (which internally contains a 'var') in a field initialzer or a this/base call.

Reply:
also, top level statements.